pages/company/visualizations/commit_domains.py

```diff
@@ -0,0 +1,212 @@
+                                        id=f"{PAGE}-date-picker-range-{VIZ_ID}",
+                                        min_date_allowed=dt.date(2005, 1, 1),
+                                        max_date_allowed=dt.date.today(),
+                                        clearable=True,
+                                    ),
+                                    width="auto",
+                                ),
+                                dbc.Col(
+                                    dbc.Button(
+                                        "About Graph",
+                                        id=f"{PAGE}-popover-target-{VIZ_ID}",
+                                        color="secondary",
+                                        size="sm",
+                                    ),
+                                    width="auto",
+                                    style={"paddingTop": ".5em"},
+                                ),
+                            ],
+                            align="center",
+                            justify="between",
+                        ),
+                    ]
+                ),
+            ]
+        )
+    ],
+)
+
+# callback for graph info popover
+@callback(
+    Output(f"{PAGE}-popover-{VIZ_ID}", "is_open"),
+    [Input(f"{PAGE}-popover-target-{VIZ_ID}", "n_clicks")],
+    [State(f"{PAGE}-popover-{VIZ_ID}", "is_open")],
+)
+def toggle_popover(n, is_open):
+    if n:
+        return not is_open
+    return is_open
+
+
+# callback for Company Affiliation by Github Account Info graph
+@callback(
+    Output(VIZ_ID, "figure"),
+    [
+        Input("repo-choices", "data"),
+        Input(f"{PAGE}-{paramter_1}-{VIZ_ID}", "value"),
+        Input(f"{PAGE}-date-picker-range-{VIZ_ID}", "start_date"),
+        Input(f"{PAGE}-date-picker-range-{VIZ_ID}", "end_date"),
+    ],
+    background=True,
+    # prevent_initial_call=True,
+)
+def commit_domains_graph(repolist, num, start_date, end_date):
+
+    # wait for data to asynchronously download and become available.
+    cache = cm()
+    df = cache.grabm(func=cq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=cq, repos=repolist)
+
+    start = time.perf_counter()
+    logging.debug(f"{VIZ_ID}- START")
+
+    # test if there is data
+    if df.empty:
+        logging.debug(f"{VIZ_ID} - NO DATA AVAILABLE")
+        return nodata_graph
+
+    # function for all data pre processing, COULD HAVE ADDITIONAL INPUTS AND OUTPUTS
+    df = process_data(df, num, start_date, end_date)
+
+    fig = create_figure(df)
+
+    logging.debug(f"{VIZ_ID} - END - {time.perf_counter() - start}")
+    return fig
+
+
+def process_data(df: pd.DataFrame, num, start_date, end_date):
+
+    # convert to datetime objects rather than strings
+    df["author_timestamp"] = pd.to_datetime(df["author_timestamp"], utc=True)
+
+    # order values chronologically by author_timestamp date earliest to latest
```

Comment:
just thought of this- we're ordering by author timestamp but then getting committer emails- maybe we should pick author or committer and stick with that?

Reply:
Oh good catch thats a mistake

Reply:
Looking into it now I was doing "c.cmt_author_email AS committer_email" in the query which makes no sense looking back lol, changing it now